pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -43,9 +43,14 @@ def __init__(self, trainer: "pl.Trainer", resume_from_checkpoint: Optional[_PATH
     @property
     def hpc_resume_path(self) -> Optional[str]:
         dir_path_hpc = str(self.trainer.weights_save_path)
+        if dir_path_hpc is None or not os.path.isdir(dir_path_hpc):
+            return None
         max_version = self.max_ckpt_version_in_folder(dir_path_hpc, "hpc_ckpt_")
         if max_version is not None:
             return os.path.join(dir_path_hpc, f"hpc_ckpt_{max_version}.ckpt")
+        if ".pl_auto_save.ckpt" in os.listdir(dir_path_hpc):
+            auto_save_checkpoint = os.path.join(dir_path_hpc, ".pl_auto_save.ckpt")
+            return auto_save_checkpoint
```

Comment:
this only takes effect for HPC resume path? why isn't it part of the general `resume_from_checkpoint` flow? 

isn't fault tolerance a superset of the HPC support currently in Lightning? Why doesn't this supersede that so we can deprecate the explicit hpc hooks?

Reply:
This isn't part of the `resume_from_checkpoint`, as users shouldn't have to modify their Trainer for their training for fault tolerant to resume. This is meant to enable auto_restart on preemptive failure.

Yes, the depreciation and modification would be done in the following PR.

Reply:
resume_from_checkpoint is really meant to reload from a specific checkpoint generated by the user (c.f ModelCheckpoint)